pkg/apis/workflow/v1alpha1/workflow_types.go

```diff
@@ -677,6 +695,12 @@ type NodeStatus struct {
 	// Not applicable to virtual nodes (e.g. Retry, StepGroup)
 	TemplateRef *TemplateRef `json:"templateRef,omitempty"`
 
+	// StoredTemplateID is an ID used to store resolved templates.
+	StoredTemplateID string `json:"storedTemplateID,omitempty"`
```

Comment:
I thought `workflowtemplatename/templatename` is not reliable because the resolved template contents can be different if passing inputs are different. However, as I set empty inputs for stored templates, it should work. Let me try it.

Reply:
Ah, no. Nested templates can be different even if the final resolved template location is the same. I'll try something like `workflowtemplatename/templatename-hash`.

Reply:
I realized I could use original template reference info as a stored base template key because if the entry template reference is the same, the resolved template should be identical.